curriculum/challenges/english/25-front-end-development/lab-moon-orbit/66a37f37ef5823a313de8c26.md

```diff
@@ -0,0 +1,355 @@
+```
+
+You should have a `div` with the class `earth` inside the `.space` `div`.
+
+```js
+assert.exists(document.querySelector('div.space div.earth'));
+```
+
+The `earth` div should be the first child of the `space` `div`.
+
+```js
+assert(document.querySelector('.space > div')?.getAttribute('class') === 'earth');
+```
+
+The `earth` `div` should have a width and height of `100px`.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.earth')?.width === '100px' && new __helpers.CSSHelp(document).getStyle('.earth')?.height === '100px');
+```
+
+The `earth` `div` should use `absolute` positioning.
+
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.earth')?.position === 'absolute');
+```
+
+The `earth` `div` should have a `top` and `left` position of `50%`.
+
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.earth')?.top === '50%' && new __helpers.CSSHelp(document).getStyle('.earth')?.left === '50%');
+```
+
+The `earth` `div` should be moved back and up by `50%` by setting the `transform` property to `translate(-50%, -50%)`.
+
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.earth')?.transform === 'translate(-50%, -50%)');
+```
+
+You should have a `div` with the class `orbit` inside the `.space` `div`.
+
+```js
+assert.exists(document.querySelector('div.space div.orbit'));
+```
+
+The `orbit` class `div` should come after the `earth` class `div`.
+
+```js
+assert(document.querySelector('.earth')?.nextElementSibling?.classList?.contains('orbit'));
+```
+
+The `orbit` `div` should have a width and height of `200px`.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.orbit')?.width === '200px' && new __helpers.CSSHelp(document).getStyle('.orbit')?.height === '200px');
+```
+
+The `orbit` `div` should use `absolute` positioning.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.orbit')?.position === 'absolute');
+```
+
+The `orbit` `div` should be positioned at the center of the `space` `div` using a `transform` property that shifts it by `-50%` on both the vertical and horizontal axes.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.orbit')?.transform === 'translate(-50%, -50%)');
+```
+
+You should have a `div` with the class `moon` inside the `.orbit` `div`.
+
+```js
+assert.exists(document.querySelector('div.space div.orbit div.moon'));
+```
+
+The `moon` `div` should have a width and height of `30px`.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.moon')?.width === '30px' && new __helpers.CSSHelp(document).getStyle('.moon')?.height === '30px');
+```
+
+The `moon` `div` should be positioned using `absolute` positioning.
+
+```js
+assert(new __helpers.CSSHelp(document).getStyle('.moon')?.position === 'absolute');
+```
+
+The `moon` `div` should position itself at the top of the `orbit` `div` and be centered horizontally.
```

Comment:
I had some difficulties to understand that. Can we reword it in some way?

Reply:
I have split this hint into two.